### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by jonstorer**
This is the legalshield strategy for passport.js. The profile conforms to the basics of the portable contacts spec. (even though the spec seems to have been dropped.)

These other strategies follow the portable contacts spec, at least some.

https://github.com/jaredhanson/passport-facebook/blob/master/lib/strategy.js#L176
https://github.com/florianheinemann/passport-dropbox-oauth2/blob/master/lib/passport-dropbox-oauth2/strategy.js#L126
https://github.com/jaredhanson/passport-twitter/blob/master/lib/strategy.js#L149
https://github.com/jaredhanson/passport-linkedin/blob/master/lib/strategy.js#L150
https://github.com/jaredhanson/passport-windowslive/blob/master/lib/strategy.js#L121
https://github.com/jaredhanson/passport-paypal-oauth/blob/master/lib/passport-paypal-oauth/strategy.js#L80
https://github.com/mjpearson/passport-slack/blob/master/lib/passport-slack/strategy.js#L98

### CODE REVIEW

File: lib/passport-legalshield-oauth2/strategy.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    OAuth2Strategy = require('passport-oauth').OAuth2Strategy,
+    InternalOAuthError = require('passport-oauth').InternalOAuthError;
+
+var Strategy = function Strategy (options, verify) {
+  options = JSON.parse(JSON.stringify(options || {}));
+
+  var userTypesMap = {
+    membership: '/v2/my/memberships',
+    associate: '/v2/my/associate',
+    admin: '/v2/my/admin'
```
Comment (douglasrlee) [inline lib/passport-legalshield-oauth2/strategy.js]: This should be `/v2/admin/me`

File: test/configSpec.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
+var LegalSheild = require('../'),
+    expect = require('chai').expect;
+
+it('exports the Strategy', function () {
+  expect(LegalSheild.Strategy).to.exist;
+});
+
+describe('config', function () {
```
Comment (douglasrlee) [inline test/configSpec.js]: Is this BDD style test chai?

File: test/configSpec.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+  describe('_profileURL', function () {
+    it('is set correctly for userType membership', function () {
+      options.userType = 'membership';
+      strategy = new LegalSheild.Strategy(options, verify);
+      expect(strategy._profileURL).to.eql('https://api.legalshield.com/v2/my/memberships');
+    });
+
+    it('is set correctly for userType associate', function () {
+      options.userType = 'associate';
```
Comment (douglasrlee) [inline test/configSpec.js]: Can you also determine the userType based on the contents of the JWT?

File: lib/passport-legalshield-oauth2/strategy.js (hunk 1 above)
Comment (jonstorer) [inline lib/passport-legalshield-oauth2/strategy.js]: Thanks! I'll make sure to get this updated.

File: test/configSpec.js (hunk 2 above)
Comment (jonstorer) [inline test/configSpec.js]: This is BDD style. I had a hard time choosing the describe topics, this is a small library, but I didn't want just one big test file.

File: test/configSpec.js (hunk 3 above)
Comment (jonstorer) [inline test/configSpec.js]: I could, that's a good point. The client is tied to a type, so I should be able to get that out of the JWT. Thanks!

File: test/configSpec.js (hunk 2 above)
Comment (douglasrlee) [inline test/configSpec.js]: 👍

File: test/configSpec.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+describe('config', function () {
+  var strategy, options, verify;
+
+  beforeEach(function () {
+    options = {
+      clientID: 'my-client-id',
+      clientSecret: 'my-client-secret'
+    };
+    verify = function () {};
+    strategy = new LegalSheild.Strategy(options, verify);
```
Comment (sftwninja) [inline test/configSpec.js]: `LegalSheild` typo?

File: test/profileSpec.js [hunk 5]
```diff
@@ -0,0 +1,286 @@
+var LegalSheild = require('../'),
```
Comment (sftwninja) [inline test/profileSpec.js]: Same typo

File: lib/passport-legalshield-oauth2/strategy.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+  this._baseURL = options.baseURL || 'https://api.legalshield.com';
+
+  options.authorizationURL = options.authorizationURL || this._baseURL + '/auth/authorize';
+  options.tokenURL         = options.tokenURL         || this._baseURL + '/auth/token';
+
+  OAuth2Strategy.call(this, options, verify);
+
+  this.name = options.name || 'legalshield-oauth2';
+};
```
Comment (julamula) [inline lib/passport-legalshield-oauth2/strategy.js]: good spelling on this one!! :)

**Pull request closed without merging**